[Profiler] Enable Iterative Step without profiler in fbcode

sraikund16 opened the pull request:
Summary: Adds post optimizer hook for fbcode so that we can run iterative on demand without having to use a frontend profiler interface. Since this is being used more frequently, it would be convenient for users to be able to trigger this on-demand feature without having to worry about being within some timing window.

Test Plan: Ran iterative tracing without profiler.profile

Differential Revision: D66734119

briancoutinho reviewed (approved):
lgmt! thanks for enabling optimizer hook setting internally

sraikund16 committed:
[Profiler] Enable Iterative Step without profiler in fbcode (#142077)

Summary:

Adds post optimizer hook for fbcode so that we can run iterative on demand without having to use a frontend profiler interface. Since this is being used more frequently, it would be convenient for users to be able to trigger this on-demand feature without having to worry about being within some timing window

Test Plan:
Ran iterative tracing without profiler.profile

Auto-trace:
https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/devvm2185.cco0.facebook.com/rank-0.Dec_10_15_45_59.1840391.pt.trace.json.gz&bucket=gpu_traces

Time-based On Demand:
https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/0/1733874778/localhost/libkineto_activities_1901872.json.gz&bucket=gpu_traces

Iteration-based On Demand without Step:
https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/0/1733875450/localhost/libkineto_activities_1976732.json.gz&bucket=gpu_traces

Reviewed By: briancoutinho, hjmeta

Differential Revision: D66734119

sraikund16 commented:
@pytorchbot merge
